<?php

namespace App\Services;

use App\Models\Medico;
use Illuminate\Validation\ValidationException;

class MedicoService
{
    public function create(array $data)
    {
        try {
            $medico = Medico::create($data);
            return $medico;
        } catch (\Exception $e) {
            throw new \Exception('Erro ao criar mÃ©dico: ' . $e->getMessage());
        }
    }

    public function update(Medico $medico, array $data)
    {
        try {
            $medico->update($data);
            return $medico;
        } catch (\Exception $e) {
            throw new \Exception('Erro ao atualizar mÃ©dico: ' . $e->getMessage());
        }
    }

    public function destroy(Medico $medico)
    {
        try {
            $medico->delete();
        } catch (\Exception $e) {
            throw new \Exception('Erro ao excluir mÃ©dico: ' . $e->getMessage());
        }
    }
}
